File: leetcode-04/solutions/cpp/1004.cpp
class Solution {
 public:
  int longestOnes(vector<int>& A, int K) {
    int i = 0;

    for (int a : A) {
      if (a == 0) K--;
      if (K < 0) {
        if (A[i] == 0) K++;
        i++;
      }
    }

    return A.size() - i;
  }
};

File: leetcode-04/solutions/python3/1004.py
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        i = 0

        for a in A:
            K -= 1 - a
            if K < 0:
                K += 1 - A[i]
                i += 1

        return len(A) - i
